File: ios/Modules/Sources/VortexCrypto/api/SecureRandomProvider.swift
import Foundation

/// Cryptographic entropy source.
///
/// A thin protocol so tests can inject deterministic bytes (known-answer
/// RNG for RFC vectors) and so platform secure RNG can be swapped for a
/// Secure Enclave-backed impl on A-series / M-series devices in a later
/// wave — the caller stays oblivious.
public protocol SecureRandomProvider: Sendable {
    func nextBytes(_ length: Int) -> Data
}

File: android/app/src/main/java/sol/vortexx/android/crypto/api/SecureRandomProvider.kt
package sol.vortexx.android.crypto.api

/**
 * Cryptographic entropy source. A thin wrapper so tests can inject a
 * deterministic impl (known-answer vectors) without touching the rest of
 * the crypto stack — and so platform SecureRandom can be swapped for a
 * hardware-backed one on Tegra / Titan M2 devices in a later wave.
 */
interface SecureRandomProvider {
    fun nextBytes(length: Int): ByteArray
}
